packages/frontend/src/pages/api/v1/posts/[postId].ts

```diff
@@ -62,7 +72,25 @@ const handleGetPost = async (req: NextApiRequest, res: NextApiResponse) => {
   `;
 
   // `upvotes` is in BigInt, so we need to convert it to a number
-  res.send(posts.map((post) => ({ ...post, upvotes: parseInt(post.upvotes) })));
+  posts = posts.map((post) => ({ ...post, upvotes: parseInt(post.upvotes) }));
+
+  // Transform the flat array of posts into a nested array of posts
+  const rootPost: GETPost = posts.find((post) => post.id === req.query.postId);
+
+  const withReplies = (parentId: string): GETPost[] => {
+    const replies = posts.filter((post) => post.parentId === parentId);
+    return replies.map((reply) => ({
+      ...reply,
+      replies: withReplies(reply.id),
+    }));
+  };
+
+  const postWithNestedReplies = {
+    ...rootPost,
+    replies: withReplies(rootPost.id),
```

Comment:
for practical purposes, the recursion will be fine, but i am curious what the response times look like at different depths/# of posts (since this may slow down the frontend if we send it all in one chunk), did ur tests reveal anything abt this?

Reply:
@amirgamil  Yeah agree that we need pagination for this. Not sure about the response time rn, but I'll look into it as we address things mentioned [here](https://www.notion.so/personae-labs/Nym-sprint-planning-8-0cf645411c3a4f44a264daab2824ec23?pvs=4#b45ce299e62546a4a301e30e63a4109a)!